metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/shell/DefaultStellarShellExecutor.java

```diff
@@ -0,0 +1,413 @@
+   * and adds it to the Stellar execution context.
+   *
+   * @return The global configuration.
+   */
+  @Override
+  public Map<String, Object> getGlobalConfig() {
+    Map<String, Object> globals;
+    Optional<Object> capability = getContext().getCapability(GLOBAL_CONFIG, false);
+    if (capability.isPresent()) {
+      globals = (Map<String, Object>) capability.get();
+
+    } else {
+      // if it does not exist, create it.  this creates the global config for the current stellar executor
+      // session only.  this does not change the global config maintained externally in zookeeper
+      globals = new HashMap<>();
+      getContext().addCapability(GLOBAL_CONFIG, () -> globals);
+    }
+
+    return globals;
+  }
+
+  @Override
+  public void assign(String variableName, Object value, Optional<String> expression) {
+
+    // perform the variable assignment
+    VariableResult varResult = VariableResult.withExpression(value, expression);
+    this.variables.put(variableName, varResult);
+
+    // notify any listeners
+    notifyVariableListeners(variableName, varResult);
+  }
+
+  @Override
+  public FunctionResolver getFunctionResolver() {
+    return functionResolver;
+  }
+
+  @Override
+  public Map<String, VariableResult> getState() {
+    return UnmodifiableMap.decorate(variables);
+  }
+
+  /**
+   * Returns all variables that have been defined.  Unlike 'getState' this unwraps
+   * the VariableResult so that we have the actual value.
+   *
+   * @return All variables that have been defined.
+   */
+  public Map<String, Object> getVariables() {
+    return Maps.transformValues(variables, (v) -> v.getResult());
+  }
+
+  @Override
+  public Context getContext() {
+    return context;
+  }
+
+  /**
+   * Creates a Zookeeper client.
+   * @param zookeeperUrl The Zookeeper URL.
+   */
+  private Optional<CuratorFramework> createZookeeperClient(Optional<String> zookeeperUrl) {
+    Optional<CuratorFramework> client = Optional.empty();
+
+    // can only create client, if have valid zookeeper URL
+    if(zookeeperUrl.isPresent()) {
+      String url = zookeeperUrl.get();
+      if (StringUtils.isNotBlank(url)) {
+
+        LOG.debug(String.format("Connecting to Zookeeper; url=%s", url));
+        CuratorFramework c = ConfigurationsUtils.getClient(url);
+        c.start();
+        client = Optional.of(c);
+      }
+    }
+
+    return client;
+  }
+
+  /**
+   * Creates a Context initialized with configuration stored in Zookeeper.
+   * @param properties Properties to configure the context.
+   * @param zkClient An optional Zookeeper client.
+   */
+  private Context createContext(Properties properties, Optional<CuratorFramework> zkClient) throws Exception {
+
+    Context.Builder contextBuilder = new Context.Builder()
+            .with(SHELL_VARIABLES, () -> variables)
+            .with(STELLAR_CONFIG, () -> properties);
+
+    // load global configuration from zookeeper
```

Comment:
the optional.ifPresent(()->) syntax is a nice style maybe

Reply:
I think it is simpler if I just use an 'if' statement.  I like things to be stupid simple.  'ifPresent' is nice when you have a 1 liner, but this takes quite a few lines of code, so I went with the 'if' conditional.  We do that in other places in the code base also.